contrib/inventory/hyperone.py

```diff
@@ -0,0 +1,215 @@
+
+        if config.has_option('hyperone', 'default_group'):
+            self.default_group = config.get('hyperone', 'default_group')
+
+        if config.has_option('hyperone', 'update_username'):
+            self.update_username = config.getboolean('hyperone', 'update_username')
+
+        if config.has_option('hyperone', 'connection_host'):
+            self.connection_host = config.get('hyperone', 'connection_host')
+
+        cache_path = os.path.expanduser(config.get('hyperone', 'cache_path'))
+        self.cache_path_inventory = os.path.join(cache_path, "ansible-hyperone.cache")
+
+        if not os.path.exists(cache_path):
+            os.mkdir(cache_path)
+
+        if config.has_option('hyperone', 'cache_max_age'):
+            self.cache_max_age = config.getint('hyperone', 'cache_max_age')
+
+    def read_env(self):
+        if 'H1_INVENTORY_TOKEN' in os.environ:
+            self.token = os.environ['H1_INVENTORY_TOKEN']
+
+    def parse_cli_args(self):
+        parser = argparse.ArgumentParser(description='Produce an Ansible Inventory file based on HyperOne')
+        parser.add_argument('--list', action='store_true', default=False,
+                            help='List VMs (default: True)')
+        parser.add_argument('--host', action='store',
+                            help='Get all the variables about a specific Virtual machine')
+        parser.add_argument('--purge-cache', action='store_true', default=False,
+                            help='Purge cache')
+        return parser.parse_args()
+
+    def push(self, my_dict, key, value):
+        my_dict[key] = my_dict.get(key, []) + [value]
+
+    def add_vm(self, vm):
+        name = vm['name']
+
+        self.inventory[vm['_id']] = [name, ]
+
+        for tag in vm.get('tag', {}).keys() or [self.default_group]:
+            self.push(self.inventory, tag, name)
+
+        self.push(self.inventory, 'hyperone', name)
+
+        hostvars = {'hyperone': vm}
+
+        if vm['private_ips']:
+            hostvars['private_ip'] = vm['private_ips'][0]
+
+        if self.connection_host:
+            if vm['private_ips'] and self.connection_host == 'private_ip':
+                hostvars['ansible_ssh_host'] = vm["private_ips"][0]
+                hostvars['ansible_host'] = vm["private_ips"][0]
+
+            if vm['public_ips'] and (self.connection_host == 'public_ip' or not vm['private_ips']):
+                hostvars['ansible_ssh_host'] = vm["public_ips"][0]
+                hostvars['ansible_host'] = vm["public_ips"][0]
+
+        if self.update_username:
+            hostvars['ansible_ssh_user'] = vm.get('data', {}).get('username', 'guru')
+
+        self.inventory["_meta"]["hostvars"][name] = hostvars
+
+    def _get_ips(self, netadps, type='public'):
+        return [ip['address']
+                for netadp in netadps if netadp['network']['type'] == type
+                for ip in netadp['ip']]
+
+    def get_inventory(self):
+        if self.cache.is_valid():
```

Comment:
With refactoring above https://github.com/ansible/ansible/pull/44287/files#r211606315 it will look like:
```python
        if self.cache.valid:
```